docs/trace/extending-the-sdk/README.md

```diff
@@ -1,12 +1,187 @@
 # Extending the OpenTelemetry .NET SDK
 
+Quick links:
+
 * [Building your own exporter](#exporter)
 * [Building your own instrumentation library](#instrumentation-library)
 * [Building your own processor](#processor)
 * [Building your own sampler](#sampler)
 * [Building your own resource detector](#resource-detector)
 * [References](#references)
 
+## Registration extension method guidance for library authors
```

Comment:
Lets move this to after all the components itself, so it can be referred to from places like line 238.

Reply:
Updated